python/cog/server/helpers.py

```diff
@@ -6,6 +6,32 @@
 from typing import Callable, Optional, Sequence, TextIO
 
 
+class HookedTextIOWrapper(io.TextIOWrapper):
+    def __init__(
+        self,
+        buffer: io.BufferedIOBase,
+        tee: bool,
+        hook: Callable[[str], None],
+    ) -> None:
+        self._hook = hook
+        self._buffer = []
+        self._tee = tee
+        super().__init__(buffer, line_buffering=True)
+
+    def write(self, data: str) -> int:
+        self._buffer.append(data)
+        if self._tee:
```

Comment:
Ah, I've just realised that supporting `self._tee == False` requires more work as this currently is relying on the fact that `super().write(data)` will call `flush` if the data contains line terminators. Maybe we just don't support turning teeing off? I don't know we have a valid use case for it.